cloud/kubernetes/helm/templates/Statefulsets-mysqlsa.yaml

```diff
@@ -0,0 +1,34 @@
+{{- if eq .Values.isType "sharedAll" }}
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "mysql.fullname" . }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      name: {{ include "mysql.fullname" . }}
+  serviceName: {{ include "mysqlservice.fullname" . }}
+  template:
+    metadata:
+      labels:
+        name: {{ include "mysql.fullname" . }}
+    spec:
+      containers:
+      - name: {{ .Values.mysql.spec.template.spec.containers.name }}
+        image: {{ .Values.mysql.spec.template.spec.containers.image }}
+        args: [ "--lower_case_table_names=1","--sql_mode=IGNORE_SPACE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION"]
+        env:
+        - name: {{ .Values.mysql.spec.template.spec.containers.env.mysqlenv1 }}
+          value: {{ .Values.mysql.spec.template.spec.containers.env.mysqlval1 }}
+        - name: {{ .Values.mysql.spec.template.spec.containers.env.mysqlenv2 }}
+          value: {{ .Values.mysql.spec.template.spec.containers.env.mysqlval2 }}
+        - name: {{ .Values.mysql.spec.template.spec.containers.env.mysqlenv3 }}
+          value: {{ .Values.mysql.spec.template.spec.containers.env.mysqlval3 }}
+        - name: {{ .Values.mysql.spec.template.spec.containers.env.mysqlenv4 }}
+          value: {{ .Values.mysql.spec.template.spec.containers.env.mysqlval4 }}
+        ports:
+        - containerPort: {{ .Values.mysql.spec.template.spec.containers.ports.containerPort }}
+{{ include "sharedAll.second" . | indent 8 }}
```

Comment:
instead of first/second/third, use appropriate names like sharedAll.config/persistentvolumes for common set of properties/attributes used across multiple yaml files.